src/itemstackmetadata.cpp

```diff
@@ -29,15 +33,21 @@ void ItemStackMetadata::deSerialize(std::istream &is)
 	m_stringvars.clear();
 
 	if (!in.empty() && in[0] == DESERIALIZE_START) {
+		legacy_metadata = "";
+
 		Strfnd fnd(in);
 		fnd.to(1);
 		while (!fnd.at_end()) {
 			std::string name = fnd.next(DESERIALIZE_KV_DELIM_STR);
 			std::string var  = fnd.next(DESERIALIZE_PAIR_DELIM_STR);
-			m_stringvars[name] = var;
+			if (name == "") {
```

Comment:
Backwards compability disallows any real change here. `set_metadata` will ALWAYS overwrites the metadata string `""` since there are mods that already saved (and still save) the metadata string into that field.
Perhaps it would be better to not change anything here.